sql/item_subselect.cc

```diff
@@ -6966,3 +6992,47 @@ void Item_subselect::init_expr_cache_tracker(THD *thd)
   DBUG_ASSERT(expr_cache->type() == Item::EXPR_CACHE_ITEM);
   node->cache_tracker= ((Item_cache_wrapper *)expr_cache)->init_tracker(qw->mem_root);
 }
+
+
+const char *Explain_subq_materialization::exec_strategy_str
+  [subselect_hash_sj_engine::exec_strategy::END_OF_EXEC_STRATEGIES + 1]=
+{
+  "undefined", "complete_match", "partial_match", "partial_match_merge",
+  "partial_match_scan", "impossible"
+};
+
+
+void Explain_subq_materialization::track_partial_merge_keys(
+    Ordered_key **merge_keys, uint merge_keys_count)
+{
+  partial_match_merge_keys_counts.resize(merge_keys_count, 0);
+  for (uint i= 0; i < merge_keys_count; i++)
+  {
+    partial_match_merge_keys_counts[i]=
+        merge_keys[i]->get_key_buff_elements();
+  }
+}
+
+
+void Explain_subq_materialization::print_explain_json(Json_writer *writer,
+                                                      bool is_analyze)
+{
+  writer->add_member("subquery_materialization").start_object();
+  if (is_analyze)
+  {
+    writer->add_member("r_exec_strategy").add_str(
+          exec_strategy_str[exec_strategy]);
+    if (partial_match_buffer_size != -1)
+    {
+      writer->add_member("r_partial_match_buffer_size").add_size(
+            partial_match_buffer_size);
+    }
+
+    for(uint i= 0; i < partial_match_merge_keys_counts.elements(); i++)
```

Comment:
Any reason for not printing a JSON array here?

Reply:
Agree, fixed in the new revision